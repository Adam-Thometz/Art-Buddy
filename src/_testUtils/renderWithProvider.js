import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;