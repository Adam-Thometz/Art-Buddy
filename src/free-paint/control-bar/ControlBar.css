.ControlBar {
  display: flex;
  flex-direction: column;
}

.ControlBar-button-wrapper {
  display: flex;
  align-items: center;
}

.ControlBar-button-wrapper button {
  z-index: 6;
  width: 105px;
  height: 90px;
  border-width: 6px;
  margin: 2px;
  padding: 3%;
}

.ControlBar-shape-button {
  width: 45px;
  height: 45px;
}