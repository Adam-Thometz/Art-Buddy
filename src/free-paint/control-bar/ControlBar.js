import React from 'react';

import { useDispatch, useSelector } from 'react-redux';

import './ControlBar.css';

import Button from '_components/button/Button';
import Icon from '_components/icon/Icon';

import square from '_media/free-paint/shapes/square.png';
import pencil from '_media/free-paint/pencil.png';
import eraser from '_media/free-paint/eraser.png';
import options from '_data/free-paint/freePaintOptions';
import colorsToFilter from '_data/free-paint/colorToFilter';

const Options = () => {
  const { color, display } = useSelector(state => state.freePaint);
  const dispatch = useDispatch()

  const handleChangeColor = e => {

  }
  
  const handleChangeDisplay = e => {

  }

  const buttonStyle = {
    width: '105px',
    height: '90px',
    borderWidth: '6px',
    margin: '2px',
    padding: '3%'
  };

  const pencilColor = { filter: colorsToFilter[color] };

  return (
    <div className='ControlBar'>
      <Button otherStyles={buttonStyle} colorId={4}>ABC</Button>
      <Button otherStyles={buttonStyle} colorId={0}>abc</Button>
      <Button otherStyles={buttonStyle} colorId={1}>123</Button>
      <Button otherStyles={buttonStyle} colorId={2}><img
        className='ControlBar-shape-button'
        src={square}
        alt="Shapes"
      /></Button>
      <Button 
        otherStyles={{ ...buttonStyle, ...pencilColor }}
      ><Icon size='45px' text="Pencil" icon={pencil} /></Button>
      <Button otherStyles={buttonStyle}><Icon size='45px' text="Eraser" icon={eraser} /></Button>
    </div>
  );
};

export default Options;