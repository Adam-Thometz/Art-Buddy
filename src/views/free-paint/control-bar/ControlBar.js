import { useSelector, useDispatch } from "react-redux";
import { setIsErasing, setEditStencilMode } from "store/free-paint/freePaintReducer";

import "./ControlBar.css";

import ButtonWrapper from "./button-wrapper/ButtonWrapper";
import StencilOptions from "./stencil-options/StencilOptions";
import ColorOptions from "./color-options/ColorOptions";

import { SQUARE_SHAPE, STENCIL, PENCIL, ERASER } from "assets/free-paint/freePaint.assets";

const ControlBar = () => {
  const { color, isEditingStencil, isErasing } = useSelector(
    (state) => state.freePaint
  );
  const dispatch = useDispatch();

  const handleSetEraser = () => dispatch(setIsErasing(!isErasing));

  const handleEditStencilMode = () =>
    dispatch(setEditStencilMode(!isEditingStencil));

  return (
    <section className="ControlBar">
      <ButtonWrapper
        label="ABC"
        id="upperCase"
        Popout={StencilOptions}
        colorId={4}
      />
      <ButtonWrapper
        label="abc"
        id="lowerCase"
        Popout={StencilOptions}
        colorId={0}
      />
      <ButtonWrapper
        label="123"
        id="numbers"
        Popout={StencilOptions}
        colorId={1}
      />
      <ButtonWrapper
        label={SQUARE_SHAPE}
        id="shapes"
        Popout={StencilOptions}
        colorId={2}
      />

      <ButtonWrapper
        label="Stencil"
        iconImg={STENCIL}
        onClick={handleEditStencilMode}
      />
      <ButtonWrapper
        label="Pencil"
        id="colors"
        iconImg={PENCIL}
        Popout={ColorOptions}
        color={color}
      />
      <ButtonWrapper
        label="Eraser"
        iconImg={ERASER}
        onClick={handleSetEraser}
        color={isErasing ? "#FF69B4" : "#000000"}
      />
    </section>
  );
};

export default ControlBar;
