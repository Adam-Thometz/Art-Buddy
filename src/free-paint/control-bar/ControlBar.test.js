import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";

import renderWithProvider from "_testUtils/renderWithProvider";

import ControlBar from "./ControlBar";

describe('ControlBar component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ControlBar />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<ControlBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows options on click and hides when clicked again', () => {
    renderWithProvider(<ControlBar />);
    const abc = screen.getByText('abc');
    userEvent.click(abc)
    expect(screen.getByTestId('stencilOptions-lowerCase')).toBeInTheDocument();
    userEvent.click(abc)
    expect(screen.queryByTestId('stencilOptions-lowerCase')).not.toBeInTheDocument();
  });

  it('shows options on click and hides when a different option is clicked', () => {
    renderWithProvider(<ControlBar />);
    const abc = screen.getByText('abc');
    userEvent.click(abc)
    expect(screen.getByTestId('stencilOptions-lowerCase')).toBeInTheDocument();
    const oneTwoThree = screen.getByText('123');
    userEvent.click(oneTwoThree)
    expect(screen.queryByTestId('stencilOptions-lowerCase')).not.toBeInTheDocument();
    expect(screen.getByTestId('stencilOptions-numbers')).toBeInTheDocument();
  });
});