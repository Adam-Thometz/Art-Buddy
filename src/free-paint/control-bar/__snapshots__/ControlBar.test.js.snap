// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlBar component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="ControlBar"
  >
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id="upperCase"
        style="border-color: #17255a;"
      >
        ABC
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id="lowerCase"
        style="border-color: #89eb0a;"
      >
        abc
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id="numbers"
        style="border-color: #ffe81a;"
      >
        123
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id="shapes"
        style="border-color: #fd4d3a;"
      >
        file
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id=""
      >
        <figure
          class="Icon"
          id=""
        >
          <img
            alt=""
            src="file"
            style="height: 45px; width: 45px; cursor: pointer;"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1rem;"
          >
            Stencil
          </figcaption>
        </figure>
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id="colors"
        style="border-color: #000000; color: rgb(0, 0, 0);"
      >
        <figure
          class="Icon"
          id="colors"
        >
          <img
            alt=""
            src="file"
            style="height: 45px; width: 45px; cursor: pointer; filter: invert(0%) sepia(0%) saturate(0%) hue-rotate(304deg) brightness(100%) contrast(102%);"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1rem;"
          >
            Pencil
          </figcaption>
        </figure>
      </button>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="Button regular"
        id=""
        style="border-color: #000000; color: rgb(0, 0, 0);"
      >
        <figure
          class="Icon"
          id=""
        >
          <img
            alt=""
            src="file"
            style="height: 45px; width: 45px; cursor: pointer; filter: invert(0%) sepia(0%) saturate(0%) hue-rotate(304deg) brightness(100%) contrast(102%);"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1rem;"
          >
            Eraser
          </figcaption>
        </figure>
      </button>
    </div>
  </section>
</DocumentFragment>
`;
